test/e2e/azure_test.go

```diff
@@ -740,6 +740,15 @@ var _ = Describe("Workload cluster creation", func() {
 				}),
 			), result)
 
+			By("attaching the cluster to azure fleet", func() {
```

Comment:
Looks like the E2E test is failing specifically on k8s version `v1.28.3`, so I'm putting the test spec before the upgrade test to address this. Will try and figure out the exact issue as this test still works locally on `v1.28.3` but not on prow.